### ISSUE TITLE
Missiles have infinite fuel

### ISSUE THREAD

**Issue by TheDestroyer111** [PapaJoesSoup/BDArmory#21]
Way #1 
Step 1. Feel the urge to launch an AGM-65 off an aircraft that is standing on the runway.
Step 2. Launch the AGM-65 missile, switch to it, and look as it bounces off the runway. (or just launch it a bit upwards like any projectile with a ballistic/falling trajectory)
Step 3. After a few minutes, if the missile doesn't go up in flames due to reentry heating, you will be able to look at the now-interstellar trajectory of the still-accelerating missile. However, you will not be able to see any flame from the rocket motor - there won't be any smoke either.
Way #2 
Step 1. Feel the urge to launch some sort of missile in space.
Step 2. Somehow get a craft with ANY missile into orbit.
Step 3. Fire the missile and immediately switch to it.
Step 4. After a few minutes, you will be able to look at the now-interstellar trajectory of the still-accelerating missile. However, you will not be able to see any flame from the rocket motor - there won't be any smoke either.

**Comment by DoctorDavinci** [comment]
Looking into this now

**Comment by DoctorDavinci** [comment]
Confirmed ... It appears that all missiles, with the exception of the AGM-114, still have thrust after their fuel should have been depleted

**NOTE:**I did not include the HE-KV 1 in this test

[**Video:**](https://youtu.be/1uoJxM3Ukns)

**Comment by SpannerMonkey** [comment]
For the record
The only difference between AGM-114 and other missiles is that it has zero cruise thrust and zero cruise time, All other BDAc missiles use cruise thrust to some extent

**Comment by gomker** [comment]
I was reading up on another thread yesterday for a mod I got some Missile rails and the dev was discussing this some time ago. Not sure if its accurate, but he did develop quite a few missiles

[Thread Post](http://forum.kerbalspaceprogram.com/index.php?/topic/118352-wip01-red-vs-bluea-bdarmory-expansion-pack-now-with-100-more-stealth/&do=findComment&comment=2151913)

>...Baha never implemented fuel consumption for missiles. As most missile's boost phase accounts for about 15-30% of a missile's mass, and the sustainer phase some 5-15%, missiles act like they're 30-40% heavier than they actually are...

>... Kerbin is 1/10th scale and 1200 m/s on Kerbin isn't 1200 m/s on Earth. Thus full scale missile performance means that at 10km altitude, one can loft an AMRAAM into a suborbital trajectory. Which I have done. Admittedly it used realistic AMRAAM stats with heating turned off as I was curious as to how it would perform, but this indicates to me that there are issues we are kind of ignoring...

**Comment by goduranus** [comment]
Referred to this from #287, do missiles get lighter as they burn fuel? This is how real missiles behave, as fuel is burned they become lighter, the impact of aerodynamics increases relative to their mass and they become more maneuverable, both in terms of turn rate and lateral acceleration.

Also noticed #21's problems as well, I assumed that they were just infinite gliding, but I only checked missiles that had a coasting thrust, so I could be wrong.

Adding to #213, I noticed that sometimes when missiles explode and take out their targets, missiles survive the detonation, and kept on going. This was observed in 0.212 with AMRAAM that had a detonation range override of 20.
